pymongo/network_layer.py

```diff
@@ -228,15 +229,19 @@ async def async_receive_data(
         else:
             read_task = asyncio.create_task(_async_receive(sock, length, loop))  # type: ignore[arg-type]
         tasks = [read_task, cancellation_task]
-        result = await asyncio.wait(tasks, timeout=timeout, return_when=asyncio.FIRST_COMPLETED)
-        if len(result[1]) == 2:
+        done, pending = await asyncio.wait(
+            tasks, timeout=timeout, return_when=asyncio.FIRST_COMPLETED
+        )
+        for task in pending:
+            task.cancel()
+        if len(done) == 0:
             raise socket.timeout("timed out")
-        finished = next(iter(result[0]))
-        next(iter(result[1])).cancel()
-        if finished == read_task:
-            return finished.result()  # type: ignore[return-value]
-        else:
-            raise _OperationCancelled("operation cancelled")
+        for task in done:
+            if task == read_task:
+                return read_task.result()
+            else:
+                raise _OperationCancelled("operation cancelled")
+        return None  # type: ignore[return-value]
```

Comment:
If we write this way we can avoid the `return None`:
```python
        if len(done) == 0:
            raise socket.timeout("timed out")
        if read_task in done:
            return read_task.result()
        raise _OperationCancelled("operation cancelled")
```

Reply:
Nice simplification, thanks!